tests/unit/test_client.py

```diff
@@ -1002,3 +1002,195 @@ def __call__(self):
     with_retry(FailerUntil(2).__call__)()
     with pytest.raises(SomeException):
         with_retry(FailerUntil(3).__call__)()
+
+
+def assert_headers_with_role(
+    headers,
+    catalog,
+    schema,
+    source,
+    user,
+    accept_encoding_value,
+    client_info_value,
+    role
+):
+    assert headers[constants.HEADER_CATALOG] == catalog
+    assert headers[constants.HEADER_SCHEMA] == schema
+    assert headers[constants.HEADER_SOURCE] == source
+    assert headers[constants.HEADER_USER] == user
+    assert headers[constants.HEADER_SESSION] == ""
+    assert headers[constants.HEADER_ROLE] == role
+    assert headers["accept-encoding"] == accept_encoding_value
+    assert headers[constants.HEADER_CLIENT_INFO] == client_info_value
+    assert len(headers.keys()) == 9
+
+
+def test_request_headers_role_hive_all(mock_get_and_post):
+    get, post = mock_get_and_post
+
+    catalog = "test_catalog"
+    schema = "test_schema"
+    user = "test_user"
+    source = "test_source"
+    accept_encoding_header = "accept-encoding"
+    accept_encoding_value = "identity,deflate,gzip"
+    client_info_header = constants.HEADER_CLIENT_INFO
+    client_info_value = "some_client_info"
+
+    req = TrinoRequest(
+        host="coordinator",
+        port=8080,
+        client_session=ClientSession(
+            user=user,
+            source=source,
+            catalog=catalog,
+            schema=schema,
+            role="hive=ALL",
+            headers={
+                accept_encoding_header: accept_encoding_value,
+                client_info_header: client_info_value,
+            }
+        ),
+        http_scheme="http",
+        redirect_handler=None,
+    )
+
+    req.post("URL")
+    _, post_kwargs = post.call_args
+    assert_headers_with_role(
+        post_kwargs["headers"],
+        catalog,
+        schema,
+        source,
+        user,
+        accept_encoding_value,
+        client_info_value,
+        "hive=ALL"
+    )
+
+    req.get("URL")
+    _, get_kwargs = get.call_args
+    assert_headers_with_role(
+        post_kwargs["headers"],
+        catalog,
+        schema,
+        source,
+        user,
+        accept_encoding_value,
+        client_info_value,
+        "hive=ALL"
```

Comment:
Please rewrite tests to focus on role change. For instance, encoding is unrelated to role tests.

Reply:
Ready